theano/tensor/basic.py

```diff
@@ -4219,6 +4219,12 @@ def roll(x, shift, axis=None):
         else:
             axis = 0
 
+    if axis < 0:
+        if x.ndim == 0:
+            axis = 0
+        else:
+            axis = axis % x.ndim
```

Comment:
I find the way you handle the axis strange. Normally, we do something like:

    if axis < 0: axis += x.ndim

This have the advantage to return an error in case like axis=-1000 when ndim = 2. With your code it work. Does it work for numpy in that case? Can you add a test for that case and adjust the code if needed?